### PR TITLE
updated rids

### PR DESCRIPTION
**Pull request by mairaw**
@Petermarcu @eerhardt @ericstj @leecow who would be the right person to review these changes to the RID topic?

If possible, I'd also like to fix issue #6238 with this PR if I can get the appropriate info for Raspberry Pi RIDs.

I've also removed some of the out-of-support OS versions.

### CODE REVIEW

**Review by leecow — COMMENTED** [review]
From a 'supported list' perspective, the changes look good. Would want dev to confirm that it aligns with what's in the code.

File: docs/core/rid-catalog.md [hunk 1]
```diff
@@ -116,22 +116,24 @@ See [Prerequisites for .NET Core on Windows](windows-prerequisites.md) for more
 - Debian
   - `debian-x64`
   - `debian.8-x64`
+  - `debian.9-x64`
```
Comment (eerhardt) [inline docs/core/rid-catalog.md]: This should have `(.NET Core 1.1 or later versions)`. We don't support `debian.9` in 1.0.

**Review by eerhardt — APPROVED** [review]
:shipit:

File: docs/core/rid-catalog.md (hunk 1 above)
Comment (mairaw) [reply docs/core/rid-catalog.md]: Thanks @eerhardt. Fixed!

**Comment by mairaw** [comment]
I also noticed that we have one more issue related to this same topic: #5731

Do we want to list the musl entries or add a note that they're not listed the same way we do with arm and corert values?

**Review by JRAlexander — APPROVED** [review]
LGTM.

**Comment by Petermarcu** [comment]
musl should be listed because Alpine is a supported platform and that is the generic C library for Alpine.

**Comment by Petermarcu** [comment]
In the long run, we should talk about the sustainability of this doc and whether it should be more generic.

**Comment by mairaw** [comment]
Thanks @Petermarcu. I'll handle that on a separate PR and follow up with you offline on making this topic more generic.

**Pull request merged**